DragaliaAPI/DragaliaAPI/Features/Missions/InitialProgress/MissionInitialProgressionService.cs

```diff
@@ -284,6 +292,27 @@ await apiContext
         return await apiContext.PlayerCharaData.MaxAsync(x => x.Level);
     }
 
+    private async Task<int> GetCharacterLevelCount(UnitElement? element, int? level)
+    {
+        if (element != null)
+        {
+            return (
+                await apiContext
+                    .PlayerCharaData.Select(x => new { x.CharaId, x.Level })
+                    .ToListAsync()
+            )
+                .Where(x => MasterAsset.CharaData[x.CharaId].ElementalType == element)
+                .Where(x => x.Level >= level)
```

Comment:
Could we bring the level condition into the query to reduce the number of rows we are retrieving?

Reply:
Not super confident with LINQ for SQL in particular, but I think the commit I just added should move the condition into the sql query

Reply:
Yeah what you've done is correct - anything before the .ToListAsync() will be used to compose the SQL query.

I noticed a few other improvements that could be made. 

- Using `.Count()` instead of `Where.Count()` is more concise, and
- in the no-element case `.ToListAsync().Count` is fetching all the row data when a `.CountAsync()` i.e. SQL `COUNT(*)` will suffice

Just pushed a commit to make those changes rather than do another round of feedback :)